config.go

```diff
@@ -541,3 +551,112 @@ type VersionMistmatchError string
 func (e VersionMistmatchError) Error() string {
 	return string(e)
 }
+
+// Restart re-run the app with all the arguments passed
+func (config *TGFConfig) Restart() int {
+	cmd := exec.Command(os.Args[0], os.Args[1:]...)
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
+	if err := cmd.Run(); err != nil {
+		printError("Error on restart: %v", err)
+		return 1
+	}
+	return 0
+}
+
+// Debug print debug information with formatting string
+func (config *TGFConfig) Debug(format string, args ...interface{}) {
+	config.tgf.Debug(format, args...)
+}
+
+// GetUpdateVersion fetches the latest tgf version number from the GITHUB_API
+func (config *TGFConfig) GetUpdateVersion() (string, error) {
+	resp, err := http.Get("https://api.github.com/repos/coveooss/tgf/releases/latest")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	var jsonResponse map[string]string
+	json.NewDecoder(resp.Body).Decode(&jsonResponse)
+	latestVersion := jsonResponse["tag_name"]
+	if latestVersion == "" {
+		return "", errors.New("Error parsing json response")
+	}
+	return latestVersion[1:], nil
+}
+
+// ShouldUpdate evaluate wether tgf updater should run or not depending on cli options and config file
+func (config *TGFConfig) ShouldUpdate() bool {
```

Comment:
You removed a case during your refactoring. When `version` is locallyBuilt, we should:
* Only update if AutoUpdate is set to true (from the CLI)
* There should be an exception in the `RunWithUpdateCheck` for that version because semver will fail

Please test before requesting a review, this should've been the first tested case